src/lib/security/selinux.rb

```diff
@@ -0,0 +1,89 @@
+# Copyright (c) [2021] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+
+module Security
+  # Class for handling SELinux kernel params
+  class Selinux
+    Yast.import "Bootloader"
+
+    MODULE_NAME = "selinux".freeze
+    private_constant :MODULE_NAME
+
+    ENABLE_VALUE = "1".freeze
+    private_constant :ENABLE_VALUE
+
+    def enabled?
+      param(:security) == MODULE_NAME && param(:selinux) == ENABLE_VALUE
+    end
+
+    def enforcing?
+      return false unless enabled?
+
+      param("enforcing") == ENABLE_VALUE
+    end
+
+    def permissive?
+      return false unless enabled?
+
+      !enforcing?
+    end
+
+    def disabled?
+      !enabled?
+    end
+
+    def enable
+      update_params("security" => MODULE_NAME, "selinux" => ENABLE_VALUE)
+    end
+
+    def enforcing
+      enable
+      update_params("enforcing" => ENABLE_VALUE)
+    end
+
+    def permissive
+      enable
+      update_params("enforcing" => :missing)
+    end
+
+    def disable
+      update_params(
+        "security"  => :missing,
+        "selinux"   => :missing,
+        "enforcing" => :missing
+      )
+    end
+
+    def save
+      Yast::Bootloader.Write
+    end
+
+    private
+
+    def param(key)
+      Yast::Bootloader.kernel_param(:common, key.to_s)
```

Comment:
be awre that kernel parameter can be also array. I am not sure how it will behave in selinux case if something is specified multiple times.
see https://github.com/config-files-api/config_files_api_grub2/blob/master/lib/cfa/grub2/default.rb#L217

I have to update also Bootloader module documentation. It was added when kernel has specific multiple output consoles.